### DISCUSSION TITLE
How to stream an mp3 file over UDP?

### DISCUSSION THREAD

**Discussion by mvoss96** [pschatzmann/arduino-audio-tools#1749] [Q&A] [answered]
Hi,
im trying to stream a mp3 file using ffmpeg from a computer to the esp32.

im using the command:
`ffmpeg -re -i example.mp3 -c copy -f mpegts udp://192.168.2.63:7000`

Im trying to recieve the stream but my audio is very distorted. So im wondering what i am doing wrong?

i also tried to convert it to pcm with `./ffmpeg -re -i example.mp3 -f wav -acodec pcm_s16le udp://192.168.2.63:7000`

but that also did not create a better result

```
#include "AudioTools.h"
#include "Communication/UDPStream.h"
#include "AudioCodecs/CodecMP3Helix.h"

#define I2S_DOUT 6
#define I2S_BCLK 4
#define I2S_LRC 5

const char *ssid = "";
const char *password = "";

AudioInfo info(44100, 2, 16);
UDPStream udp(ssid, password);
const int udpPort = 7000;
VolumeStream volume;
MP3DecoderHelix codec;
EncodedAudioStream dec(&codec);
Pipeline pipeline;
I2SStream out; 
StreamCopy copier(pipeline, udp);

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // start UDP receive
  udp.begin(udpPort);

  // start I2S
  Serial.println("starting I2S...");
  // setup output
  auto config = out.defaultConfig(TX_MODE);
  // you could define e.g your pins and change other settings
  config.pin_ws = I2S_LRC;
  config.pin_bck = I2S_BCLK;
  config.pin_data = I2S_DOUT;
  config.copyFrom(info);
  config.buffer_size = 1024*16;
  config.buffer_count = 200;

  out.begin(config);
  volume.setVolume(0.2);

  // setup pipeline
  pipeline.add(dec);
  pipeline.add(volume);
  pipeline.setOutput(out);
  pipeline.begin();  // start all components with their default values


  Serial.println("started...");
}

void loop() {
  copier.copy();
}
```

**Comment by pschatzmann** [comment, marked as answer]
My guess would be that the audio is just sent too quick and therefore gets lost because there is no synchronization.
I suggest you try with IP instead of UDP.

You could try to write into a Queue and play back from the queue in a separate task as [described here](https://github.com/pschatzmann/arduino-audio-tools/wiki/Multicore-Processing)

Or better, if you are in linux, try to use [Snapcast](https://github.com/pschatzmann/arduino-snapclient)

**Reply by mvoss96** [reply]
thanks.
i got a good result using aac: `ffmpeg -re -i example.mp3 -filter:a "volume=1.5" -c:a aac -b:a 128k -ac 1 -f adts udp://192.168.2.63:7000`
But i will also checkout snapcast
